- nvcc
    - 有可能的路径是 `~/anaconda3/bin/nvcc`

## 显卡是否在槽位（显卡有没有掉）

In [1]:
!lspci | grep -i nvidia

18:00.0 VGA compatible controller: NVIDIA Corporation Device 2684 (rev a1)
18:00.1 Audio device: NVIDIA Corporation Device 22ba (rev a1)
8a:00.0 VGA compatible controller: NVIDIA Corporation Device 2684 (rev a1)
8a:00.1 Audio device: NVIDIA Corporation Device 22ba (rev a1)


- 一共2个GPU，每个GPU都有一个相应的VGA兼容控制器和一个音频设备。
    - NVIDIA Corporation Device 2684: 4090
        - NVIDIA Corporation Device 2204: 3090
    - https://admin.pci-ids.ucw.cz//mods/PC/10de/
- 18:00.0: PCI 总线（bus）
- rev al: rev，revision，硬件的修订标识符
    - `rev ff`: 有时并不表示传统意义上的硬件版本或修订
        - 硬件故障或通信问题
        - 设备未正确安装或识别
        - 备处于省电模式或未激活状态

## 驱动安装

- software update
    - `sudo systemctl disable --now unattended-upgrades`
- sudo apt install nvidia-driver-xxx
    - `ubuntu-drivers devices`
- 安装包安装：`xx.run`

## NVIDIA-SMI has failed because it couldn't communicate with the NVIDIA driver. Make sure that the latest NVIDIA driver is installed and running.

- 运行 `nvidia-smi` 命令时报错；